pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -715,54 +772,59 @@ class CreateVmModal extends React.Component {
                     providerName={providerName}
                     source={this.state.source}
                     sourceType={this.state.sourceType}
+                    os={this.state.os}
+                    osInfoList={this.props.osInfoList}
+                    downloadOSSupported={this.isVirtInstallGreaterThan220()}
                     onValueChanged={this.onValueChanged}
                     validationFailed={validationFailed} />
 
                 <hr />
 
-                { this.state.sourceType != EXISTING_DISK_IMAGE_SOURCE &&
+                {this.state.sourceType != DOWNLOAD_AN_OS &&
                 <>
-                    <StorageRow
-                        connectionName={this.state.connectionName}
-                        storageSize={this.state.storageSize}
-                        storageSizeUnit={this.state.storageSizeUnit}
+                    <OSRow
```

Comment:
So I wanted it to be clear that in case DOWNLOAD_AN_OS is the source type the source is the OS otherwise the OS choice is just an additional field.
The code difference is only the <hr> present in the latter case between the source and OS fields.

Reply:
Sorry for late answer, but wouldn't it be better to have only <hr> conditional? With just single calling of `OSRow` component. Like:
```
                <SourceRow
                    connectionName={this.state.connectionName}
                    networks={networks.filter(network => network.connectionName == this.state.connectionName)}
                    nodeDevices={nodeDevices.filter(nodeDevice => nodeDevice.connectionName == this.state.connectionName)}
                    providerName={providerName}
                    source={this.state.source}
                    sourceType={this.state.sourceType}
                    os={this.state.os}
                    osInfoList={this.props.osInfoList}
                    downloadOSSupported={this.isVirtInstallGreaterThan220()}
                    onValueChanged={this.onValueChanged}
                    validationFailed={validationFailed} />
                {this.state.sourceType != DOWNLOAD_AN_OS &&
                <hr />}

                <OSRow
                    os={this.state.os}
                    osInfoList={this.props.osInfoList}
                    onValueChanged={this.onValueChanged}
                    isLoading={this.state.autodetectOSInProgress}
                    validationFailed={validationFailed} />
                <hr />
```